frontend/src/v5/ui/routes/viewer/tickets/ticketsList/ticketsList.component.tsx

```diff
@@ -45,13 +45,20 @@ export const TicketsList = ({ tickets }: TicketsListProps) => {
 	const availableTemplatesIds = uniq(tickets.map(({ type }) => type));
 	const availableTemplates = templates.filter(({ _id }) => availableTemplatesIds.includes(_id));
 	const queries = TicketsCardHooksSelectors.selectFilteringQueries();
+	const selectedPinId = TicketsCardHooksSelectors.selectSelectedTicketPinId();
+	const isShowingPins = TicketsCardHooksSelectors.selectIsShowingPins();
 
 	const filteredByQueriesAndCompleted = TicketsCardHooksSelectors.selectTicketsFilteredByQueriesAndCompleted();
 	const filteredItems = TicketsCardHooksSelectors.selectTicketsWithAllFiltersApplied();
 
 	const toggleTemplate = (templateId: string) => TicketsCardActionsDispatchers.setTemplateFilters(xor(selectedTemplates, [templateId]));
 	const onQueriesChange = (newQueries) => TicketsCardActionsDispatchers.setQueryFilters(newQueries);
 
+	useEffect(() => {
+		if (!selectedPinId) return;
+		ViewerService.setSelectionPin({ id: selectedPinId, isSelected: isShowingPins });
+	}, [selectedPinId, isShowingPins]);
```

Comment:
I dont get what this code is for

Reply:
Without this selected pins are not reselected when isShowingPins changes. So you may have a list item selected, but its corresponding pin won't be selected in the viewer. I also noticed a similar bug to do with sequences showing/hiding pins (https://github.com/3drepo/3D-Repo-Product-Team/issues/572)

Reply:
This is more like a patch instead of a solution. The problem is not here, is in `src\v4\routes\viewer3D\viewer3D.component.tsx`.

If you take a look at this code:

<img width="539" alt="image" src="https://github.com/user-attachments/assets/ff3b20b4-8427-4414-9dcb-17dd960bb4b9">

Because with Promise.all it tries to execute all of the promises at the same time, changing the selection state of a pin can fail. For example if you try to change the select status of a pin that hasnt been added to the viewer yet.

This guarantees that the selection changes are applied after the pins has been added to the viewer:
<img width="595" alt="image" src="https://github.com/user-attachments/assets/435a243c-1263-4988-8ee6-726fa5e6613d">

So remove this bit of code from here and make the change in viewer3D. It should fix the other problem with sequences  unless there is another bit that can add to the bug.